platform_plugin_aspects/xblock.py

```diff
@@ -180,3 +187,27 @@ def _get_statici18n_js_url():
             ):
                 return text_js.format(locale_code=code)
         return None
+
+    @XBlock.json_handler
+    def get_superset_guest_token(
+        self, request_body, suffix=""
+    ):  # pylint: disable=unused-argument
+        """Return a guest token for Superset."""
+        user_service = self.runtime.service(self, "user")
+        user = user_service.get_current_user()
+
+        try:
+            guest_token = generate_guest_token(
+                user=user,
+                course=self.runtime.course_id,
+                dashboards=self.dashboards(),
+                filters=self.filters,
```

Comment:
I have some security concerns here, in case a user knows the dashboard UUID (which is public anyway) and uses the SupersetXBlock, as the only filters applied are what the user has defined, they would be able to see information for all courses.

@bmtcril Not sure what we can do here

Reply:
The [view prepends the built-in course filters](https://github.com/openedx/platform-plugin-aspects/pull/27/files/569f3952d7aad87c879778f8793af9ed812b2b01#diff-1edce6254e2a9acff5c08fb9bd8d45316b5b9a287502112014844d3faa6344e2R131) before generating the user token, so I think we're ok here?